apps/api/src/pages/pages.service.ts

```diff
@@ -0,0 +1,767 @@
+              INSERT INTO "page_node_relations" (
+                "relation_id", "page_id", "node_id", "node_type", "entity_id", 
+                "order_index", "node_data", "created_at", "updated_at"
+              )
+              VALUES (
+                ${relationId}, ${pageId}, ${nodeId}, ${nodeType}, ${entityId}, 
+                ${orderIndex}, ${nodeDataStr}, NOW(), NOW()
+              )
+              RETURNING *
+            `;
+          return created;
+        });
+
+        updatedRelations = await Promise.all(updatedRelationsPromises);
+      }
+
+      // Handle non-database operations after transaction (like updating page state file)
+      if (pageConfig) {
+        try {
+          // Read existing state file
+          let ydoc = new Y.Doc();
+
+          if (page.state_storage_key) {
+            try {
+              const stateStream = await this.minio.client.getObject(page.state_storage_key);
+              const stateBuffer = await streamToBuffer(stateStream);
+
+              if (stateBuffer) {
+                const update = new Uint8Array(stateBuffer);
+                Y.applyUpdate(ydoc, update);
+              }
+            } catch (error) {
+              // If file doesn't exist, use new Y.Doc()
+              this.logger.error('Error reading state file, creating new:', error);
+              ydoc = new Y.Doc();
+            }
+          }
+
+          // Update pageConfig
+          const pageConfigMap = ydoc.getMap('pageConfig');
+          for (const [key, value] of Object.entries(pageConfig)) {
+            pageConfigMap.set(key, value);
+          }
+
+          // Save updated state
+          const update = Y.encodeStateAsUpdate(ydoc);
+          await this.minio.client.putObject(page.state_storage_key, Buffer.from(update));
+        } catch (error) {
+          this.logger.error('Error saving page state:', error);
+        }
+      }
+
+      return {
+        page: updatePage,
+        nodeRelations: updatedRelations,
+      };
+    });
+  }
+
+  // Share page - create public share link
+  async sharePage(user: User | ResolveUserResponse, pageId: string): Promise<SharePageResult> {
+    const { uid } = user as User;
+
+    // Check if page exists
+    const [page] = await this.prisma.$queryRaw<Page[]>`
+      SELECT * FROM "pages"
+      WHERE "page_id" = ${pageId}
+      AND "uid" = ${uid}
+      AND "deleted_at" IS NULL
+    `;
+
+    if (!page) {
+      throw new PageNotFoundError();
+    }
+
+    // Use ShareService to create share
+    const shareRecord = await this.shareService.createShare(user as User, {
+      entityId: pageId,
+      entityType: 'page' as EntityType,
+      title: page.title,
+      allowDuplication: false, // Default: do not allow duplication
+    });
+
+    return {
+      pageId,
+      canvasId: page.canvas_id,
+      shareId: shareRecord.shareId,
+      shareUrl: `${process.env.FRONTEND_URL || 'https://refly.ai'}/share/pages/${shareRecord.shareId}`,
```

Comment:
Consider returning the `shareId` and let the web code decide the actual `shareUrl`